### PR TITLE
test(output): update tests to use libtest.h (fix #197)

### LINKED ISSUE #197: Update test program
[Issuehunt badges](https://issuehunt.io/r/lc-soft/LCUI/issues/197)

**Is your feature request related to a problem? Please describe.**

The output of the test program should be in the same format.

**Describe the solution you'd like**

Update the following functions in test.c:

https://github.com/lc-soft/LCUI/blob/b87ce34be7a5f4e0cd73da5a2d37d34068bfeb1c/test/test.c#L14-L26

Use the functions in libtest.c instead of `TEST_LOG()`, `CHECK()`, `CHECK_WITH_TEXT()`, just like this:

https://github.com/lc-soft/LCUI/blob/b87ce34be7a5f4e0cd73da5a2d37d34068bfeb1c/test/test_css_parser.c#L114-L145

All output should be formatted like this:

image

**Describe alternatives you've considered**
None.

**Additional context**
None.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $40.00)

- [lc-soft lc-soft](https://issuehunt.io/u/lc-soft) ($40.00)

### Submitted pull Requests
- [#208 test(output): update tests to use libtest.h (fix #197)](https://issuehunt.io/r/lc-soft/LCUI/pull/208)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/lc-soft/LCUI/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by jduo**
Change tests to use libtest.h as specified in
Issue #197.

Fixes #197 .

Changes proposed in this pull request:
- Use methods defined in libtest.h to improve test output formatting.

@lc-soft

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#197: Update test program](https://issuehunt.io/repos/5293802/issues/197)
---
</details>

### CODE REVIEW

File: test/test.c [hunk 1]
```diff
@@ -11,20 +11,19 @@ int main(void)
...
-	ret += test_textedit();
-	PRINT_TEST_RESULT(ret);
+	describe("test charset", test_charset);
+	describe("test linkedlist", test_linkedlist);
+	describe("test string", test_string);
+	describe("test strpool", test_strpool);
+	describe("test object", test_object);
+	describe("test thread", test_thread);
+	describe("test font_load", test_font_load);
+	describe("test image_reader", test_image_reader);
```
Comment (lc-soft) [inline test/test.c:21]: Underscores should be replaced with spaces

```diff
- font_load
+ font load
```

File: test/test_font_load.c [hunk 2]
```diff
@@ -4,153 +4,139 @@
...
-	CHECK( LCUIFont_LoadFile( "C:/windows/fonts/segoeuil.ttf" ) == 0 );
-	CHECK( LCUIFont_LoadFile( "C:/windows/fonts/segoeuisl.ttf" ) == 0 );
-	CHECK( LCUIFont_LoadFile( "C:/windows/fonts/segoeuiz.ttf" ) == 0 );
-	CHECK( LCUIFont_LoadFile( "C:/windows/fonts/seguibl.ttf" ) == 0 );
-	CHECK( LCUIFont_LoadFile( "C:/windows/fonts/seguili.ttf" ) == 0 );
-	CHECK_WITH_TEXT(
-		"check get font (style: normal, weight: normal)",
-		id = GetSegoeUIFont( FONT_STYLE_NORMAL, FONT_WEIGHT_NORMAL )
-	);
+	it_i("Load segoeui.ttf", LCUIFont_LoadFile( "C:/windows/fonts/segoeui.ttf" ), 0 );
```
Comment (lc-soft) [inline test/test_font_load.c:19]: Please use `clang-format` to format the code

**Comment by lc-soft** [comment]
> Note that TEST_LOG is still used in a few places
> where we log purely informational messages that
> don’t match functions in libtest.h

For code that uses TEST_LOG like below, , you can replace `TEST_LOG` with `Logger_Debug`.

https://github.com/lc-soft/LCUI/blob/4e6e459cd21f5281d13848a0cc80189e1c145618/test/test_thread.c#L19-L38

File: test/test_font_load.c (hunk 2 above)
Comment (jduo) [reply test/test_font_load.c]: I've used clang-format on all affected c files.

File: test/test.c (hunk 1 above)
Comment (jduo) [reply test/test.c]: Done. Verified all describe() calls.

**Comment by jduo** [comment]
> > Note that TEST_LOG is still used in a few places
> > where we log purely informational messages that
> > don’t match functions in libtest.h
> 
> For code that uses TEST_LOG like below, you can replace `TEST_LOG` with `Logger_Debug`.
> 
> https://github.com/lc-soft/LCUI/blob/4e6e459cd21f5281d13848a0cc80189e1c145618/test/test_thread.c#L19-L38

I've replaced calls to TEST_LOG with Logger_Debug. I've also deleted the now unused macros in test.h

**Comment by jduo** [comment]
> The output text style should be consistent, I suggest you change them to lower case.
> 
> ```diff
> - Check *****
> + check *****
> ```
> 
> image
Done.

**Pull request merged**